File: 69. Sqrt of x/main.cpp
class Solution {
public:
    int mySqrt(int x) {
        int left = 0;
        int right = x;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            long long square = (long long)mid * (long long)mid;

            if (square == (long long)x) {
                return mid;
            } else if (square < (long long)x) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return right;
    }
};

class Solution {
public:
    int mySqrt(int x) {
        long long i = 0;

        while (i * i <= (long long)x) {
            i++;
        }

        return (int)(i - 1);
    }
};

File: 69. Sqrt of x/main.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x

        while left <= right:
            mid = left + (right - left) // 2
            square = mid**2

            if square == x:
                return mid
            elif square < x:
                left = mid + 1
            else:
                right = mid - 1

        return right


class Solution:
    def mySqrt(self, x: int) -> int:
        i = 0

        while i**2 <= x:
            i += 1

        return i - 1
